driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -172,6 +172,11 @@ public static ArrayExpression<StringExpression> ofStringArray(final String... ar
         return new MqlExpression<>((cr) -> new AstPlaceholder(new BsonArray(result)));
     }
 
+    public static <R extends DocumentExpression> R current() {
```

Comment:
Why do we need a generic parameter here?  Do we expect `DocumentExpression` to be sub-typed?

Reply:
Right. The sub-type would allow a helper class (aka schema class, binding glass) to condense:
`doc.getDocument("c").<IntegerExpression>getArray("a").first()`
to
`doc.c.a.first()`